add CGit compat test case for sign/verify commit

castedo opened the pull request.

jelmer commented on tests/compat/test_porcelain.py:
```diff
@@ -91,3 +91,54 @@ def test_verify(self):
+
+
+@skipIf(
+    platform.python_implementation() == "PyPy" or sys.platform == "win32",
```
please make this dependent on gpgme actually being importable. Even on linux people might not have it installed and if we do manage to install gpgme on mac or win32 in the ci, we wouldn't have to manually update this code.

castedo replied:
I believe it is. CommitCreateSignTestCase inherits from PorcelainGpgTestCase which has a decorator
```
@skipIf(gpg is None, "gpg is not available")
```
in `dulwich/tests/test_porcelain.py`.

I don't 100% understand these `@skipIf` decorators but I suspect the current code is doing what you want.

jelmer commented on tests/compat/test_porcelain.py:
```diff
@@ -91,3 +91,54 @@ def test_verify(self):
+    "gpgme not easily available or supported on Windows and PyPy",
+)
+class CommitCreateSignTestCase(PorcelainGpgTestCase, CompatTestCase):
+    def setUp(self):
```
This setUp() is a noop, remove it?

castedo replied:
I'm not sure. I don't know Python multiple inheritance. I don't know what the default behavior is. I'm guessing `super()` only accesses the first parent class. Do you know?

You're thinking it's a noop because it just re-implements the default behavior?

castedo replied:
BTW, this setUp is a copy of the tag test case above it. I'm just copying the pattern of the existing test case for a tag, but re-adapting it to a commit.

jelmer replied:
See https://docs.python.org/3/library/functions.html#super, I'm pretty sure dropping this is a noop.

I've fixed both of these things in main for the existing test; see https://github.com/jelmer/dulwich/pull/1439

jelmer reviewed (approved):
LGTM, two minor comments.

castedo commented:
@jelmer What the heck is up with ruff? The error it triggers looks bogus. The names ruff claims are not defined are defined. They are imported in the same file.

jelmer commented:
This is because I removed the imports in main when I changed the behaviour for the existing tag tests.

If you rebase, you should see it locally.

castedo commented:
> @castedo This is because I removed the imports in main when I changed the behaviour for the existing tag tests.
> 
> If you rebase, you should see it locally.

Ah, I see now. I was under the mistaken impression that the Git Action checks run on the branch of the PR, but in fact they are running on a merge of the PR branch and the main/master branch.
